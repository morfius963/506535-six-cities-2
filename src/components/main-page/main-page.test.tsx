import * as React from "react";
import * as renderer from "react-test-renderer";

import MainPage from "./main-page";
import Cities from "../cities/cities";
import MainHeader from "../main-header/main-header";
import OffersList from "../offers-list/offers-list";
import fixtureData from "../../__fixtures__/offers";

jest.mock(`../offers-list/offers-list`, () => jest.fn().mockReturnValue(null));
jest.mock(`../main-header/main-header`, () => jest.fn().mockReturnValue(null));
jest.mock(`../cities/cities`, () => jest.fn().mockReturnValue(null));

describe(`snapshot test`, () => {
  it(`App correctly renders`, () => {
    const activeOffers = fixtureData.filter((offer) => offer.city.name === `Amsterdam`);
    const props = {
      allCities: [`Amsterdam`, `Paris`, `Hamburg`],
      city: `Amsterdam`,
      activeOffers,
      activeSort: `Popular`,
      userData: {
        email: `morf@gmail.com`
      },
      isAuthorizationRequired: false,
      onFavoriteCardToggle: jest.fn(),
      onOffersSort: jest.fn(),
      onCityClick: jest.fn(),
    };

    const tree = renderer
      .create(<MainPage {...props} />)
      .toJSON();

    expect(Cities).toHaveBeenCalled();
    expect(OffersList).toHaveBeenCalled();
    expect(MainHeader).toHaveBeenCalled();
    expect(tree).toMatchSnapshot();
  });
});
