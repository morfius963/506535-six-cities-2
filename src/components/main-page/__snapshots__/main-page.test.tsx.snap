// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test App correctly renders 1`] = `
<div
  className="page page--gray page--main"
>
  <main
    className="page__main page__main--index "
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    />
  </main>
</div>
`;
